webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceService.scala

```diff
@@ -227,4 +232,22 @@ class DataSourceService @Inject()(
     }
   }
 
+  def invalidateVaultCache(dataSource: InboxDataSource, dataLayerName: String): Fox[Unit] =
+    for {
+      genericDataSource <- dataSource.toUsable
+      dataLayer <- genericDataSource.getDataLayer(dataLayerName)
+      magsOpt = dataLayer match {
+        case layer: N5Layer          => Some(layer.mags)
+        case layer: PrecomputedLayer => Some(layer.mags)
+        case _: SegmentationLayer    => None
+        case _: WKWLayer             => None
```

Comment:
Is there a reason to have these here? Shouldn’t they match to the `_` below? Also, SegmentationLayers may be ZarrLayers at the same time, I think

Reply:
This was the automatic IntelliJ exhaustive match, will be adjusted.